watchtower/wtclient/manager.go

```diff
@@ -431,6 +436,55 @@ func (m *Manager) RemoveTower(key *btcec.PublicKey, addr net.Addr) error {
 	return nil
 }
 
+// DeactivateTower sets the given tower's status to inactive so that it is not
+// considered for session negotiation. Its sessions will also not be used while
+// the tower is inactive.
+func (m *Manager) DeactivateTower(key *btcec.PublicKey) error {
+	// We'll load the tower in order to retrieve its ID within the database.
+	tower, err := m.cfg.DB.LoadTower(key)
+	if err != nil {
+		return err
+	}
+
+	m.clientsMu.Lock()
+	defer m.clientsMu.Unlock()
+
+	for _, client := range m.clients {
+		err := client.deactivateTower(tower.ID, tower.IdentityKey)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Finally, mark the tower as inactive in the DB.
+	err = m.cfg.DB.DeactivateTower(key)
+	if err != nil {
+		log.Errorf("Could not deactivate the tower. Re-activating")
```

Comment:
nit: Do we want to log the error that caused the re-activation? Or will that be logged further up since we return the error?